backend/src/main/java/ai/verta/modeldb/entities/DatasetEntity.java

```diff
@@ -28,7 +30,10 @@ public DatasetEntity(Dataset dataset) throws InvalidProtocolBufferException {
     setName(dataset.getName());
     setDescription(dataset.getDescription());
     setTags(RdbmsUtils.convertTagListFromTagMappingList(this, dataset.getTagsList()));
-    setDataset_visibility(dataset.getDatasetVisibilityValue());
+    setDataset_visibility(
+        App.getInstance().getPublicSharingEnabled()
+            ? DatasetVisibility.PUBLIC_VALUE
+            : dataset.getDatasetVisibilityValue());
```

Comment:
We are handling logic at DAO level so this is not required because it will redundant check and leads the inconsistency in logic.

Reply:
removed